qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -282,6 +287,111 @@ def clear(self) -> None:
         self.text.get()  # also update the parameter value
 
 
+class TimeTrace(ParameterWithSetpoints):
+    """
+    A parameter class that holds the data for a time trace type measurement,
+    i.e. a measurement of N voltage or current values measured at fixed time
+    intervals
+    """
+
+    def __init__(self, name: str, instrument: Instrument, **kwargs):
+
+        self.instrument: Instrument  # needed for mypy
+        super().__init__(name=name, instrument=instrument, **kwargs)
+
+    def _validate_dt(self) -> None:
+        """
+        Validate that the specified dt (measurement time interval) can be
+        realized by the instrument with the present settings
+
+        Raises:
+            RuntimeError: If the present measurement settings prevent reaching
+                the specified dt.
+        """
+
+        minimum_time = self.instrument.sample.timer_minimum()
+        dt = self.instrument.timetrace_dt()
+        if  dt < minimum_time:
+            raise RuntimeError(f"Can not realize a time trace dt of {dt} s "
+                               f"With the present settings. The minimal "
+                               f"allowed dt is {minimum_time} s. To allow "
+                               "for a shorter dt, try changing the NPLC or "
+                               "aperture time.")
+
+
+    def _set_units_and_labels(self) -> None:
+        """
+        A helper function to set correct units and labels for the parameter
+        """
+
+        units_and_labels = {'AC Voltage': ('V', 'Voltage'),
+                            'DC Voltage': ('V', 'Voltage'),
+                            'AC Current': ('A', 'Current'),
+                            'DC Current': ('A', 'Current'),
+                            '2 Wire Resistance': ('Ohm', 'Resistance'),
+                            '4 Wire Resistance': ('Ohm', 'Resistance')}
+
+        conf = self.instrument.sense_function()
+        self.unit, self.label = units_and_labels[conf]
+
+    def _acquire_time_trace(self) -> np.ndarray:
+        """
+        The function that prepares the measurement and fetches the data
+        """
+        dt = self.instrument.timetrace_dt()
+        npts = self.instrument.timetrace_npts()
+        meas_time = npts*dt
+        disp_text = f"Acquiring {npts} samples"  # display limit: 40 characters
+        new_timeout = max(1.25 * meas_time, self.instrument.timeout())
+
+        param_settings = [(self.instrument.trigger.count, 1),
+                          (self.instrument.trigger.source, "BUS"),
```

Comment:
Do we really want to limit the use of this parameter to self triggering use case? If so, i don't mind, but this has to be documented in the code and in the example notebook.

Reply:
I'm not completely sure what you mean. Let me know if you think the notebook sufficiently documents this fact.